src/modules/systemlib/mixer/mixer_multirotor.cpp

```diff
@@ -372,6 +373,18 @@ MultirotorMixer::mix(float *outputs, unsigned space, uint16_t *status_reg)
 			     yaw * _rotors[i].yaw_scale +
 			     thrust + boost;
 
+		/*
+			implement simple model for static relationship between applied motor pwm and motor thrust
+			model: thrust = (1 - _thrust_factor) * PWM + _thrust_factor * PWM^2
+			this model assumes normalized input / output in the range [0,1] so this is the right place
+			to do it as at this stage the outputs are in that range.
+		 */
+		if (_thrust_factor > 0.0f) {
+			outputs[i] = -(1.0f - _thrust_factor) / (2.0f * _thrust_factor) + sqrtf((1.0f - _thrust_factor) *
+					(1.0f - _thrust_factor) / (4.0f * _thrust_factor * _thrust_factor) + (outputs[i] < 0.0f ? 0.0f : outputs[i] /
+							_thrust_factor));
+		}
```

Comment:
@tumbili Sorry I left this hanging for a bit, but I was vacationing for a while. 

I was misreading that line of code: it simplifies to:
const1 + sqrt( const2 + PWM/const3 ) if _thrust_factor is a constant. 

It seems that
(1.0f - _thrust_factor) / (2.0f * _thrust_factor)
and
(1.0f - _thrust_factor) * (1.0f - _thrust_factor) / (4.0f * _thrust_factor * _thrust_factor)
are also constants that could be computed only when _thrust_factor is changed.